mod/state-transition/pkg/core/state/statedb.go

```diff
@@ -283,21 +293,30 @@ func (s *StateDB[
 			withdrawalIndex++
 		}
 
-		// Cap the number of withdrawals to the maximum allowed per payload.
-		//#nosec:G701 // won't overflow in practice.
-		if len(withdrawals) == int(s.cs.MaxWithdrawalsPerPayload()) {
-			break
-		}
-
 		// Increment the validator index to process the next validator.
-		validatorIndex = (validatorIndex + 1) % math.ValidatorIndex(
-			totalValidators,
-		)
+		validatorIndex++
+		validatorIndex %= math.ValidatorIndex(totalValidators)
```

Comment:
q: I think the original was closer to specs? I am fine with this kind of changes, just what is the preference with BeaconKit (more or less concise)?

Reply:
yeah originally was exactly from spec I can keep it that way, minimize the diff for now